lib/types.ts

```diff
@@ -73,6 +77,11 @@ export interface ScanResult {
   facts: Fact[];
 }
 
+export interface AutoDetectedUserInstructions {
```

Comment:
Can we not call it "user instructions" please? I get that's how it's already called but I don't want us doubling down on it. this term doesn't exist

Reply:
Changing `AutoDetectedUserInstructions` -> `UserInstructions` seems ok in terms of domain.

Given that the interface would be no longer tied to the feature, I think it would make sense to change 
```typescript
export interface DockerFileAnalysis {
  baseImage?: string;
  dockerfilePackages: DockerFilePackages;
  dockerfileLayers: DockerFileLayers;
}
```
to
```typescript
import { UserInstructions } from "../types";

export interface DockerFileAnalysis extends UserInstructions {
  baseImage?: string;
}
```
because Dockerfile conceptually contains user instructions. WDYT @Shesekino?

Reply:
"user instructions" is a term we made up, that's not a thing.
these are instructions, or commands, that were used to build the image

Reply:
https://github.com/opencontainers/image-spec/blob/master/config.md
the spec refers to those as ```created_by```

```
created_by string, OPTIONAL

The command which created the layer.
```

Reply:
(essentially I prefer something that doesn't include ```user``` in it)

Reply:
Trying to think of a name that represents this collection of data. [reference](https://github.com/snyk/snyk-docker-plugin/blob/master/test/lib/dockerfile.test.ts#L138-L153)

```typescript
        dockerfilePackages: {
          bash: {
            instruction:
              "RUN apk update && apk upgrade && apk add --update --no-cache nodejs bash",
          },
          nodejs: {
            instruction:
              "RUN apk update && apk upgrade && apk add --update --no-cache nodejs bash",
          },
        },
        dockerfileLayers: {
          UlVOIGFwayB1cGRhdGUgJiYgYXBrIHVwZ3JhZGUgJiYgYXBrIGFkZCAtLXVwZGF0ZSAtLW5vLWNhY2hlIG5vZGVqcyBiYXNo: {
            instruction:
              "RUN apk update && apk upgrade && apk add --update --no-cache nodejs bash",
          },
        },
```

Train of thought, this structure represents how dependencies were brought into the:
* `autoDetectedUserInstructions` case:  container image's last x hours of layer(s)
* `dockerfileAnalysis` case: dockerfile's current set of instructions

The dependencies introduced by both of these cases are driven exclusively by the end user, explicitly excluding the instructions inherited from the base image the end user is using.

The long name I would give this is "how dependencies where introduced into the layers created by the end user".

What would be the short name?

Reply:
This structure represents processed user/end user instructions where the processed results includes which instructions introduced each dependency.

I honestly think `UserInstructions` or `ProcessedUserInstructions` would be appropriate names.

As you pointed out, AutoDetected should be removed from the name because the method of producing the user instructions should not be captured in the interface.

Reply:
I don't understand who is this user. These are the commands that created the layer

Reply:
some reference https://snyk.slack.com/archives/C011VKVL015/p1614009101001800?thread_ts=1614008803.001500&cid=C011VKVL015
https://snyk.slack.com/archives/C01EPSKVC9L/p1613997226010300?thread_ts=1613992505.006200&cid=C01EPSKVC9L

Reply:
I suggest someone from Capsule take over for finding the terminology we want to use - it would be their responsibility to pay back that tech debt eventually (or handle the consequences of not)

Reply:
> I don't understand who is this user. These are the commands that created the layer

I've believe i've just understood your viewpoint. These are the Dockerfile instructions or approximated Dockerfile used to build the top most layer.

Reply:
Correct, for the most part, I think :)
Are they necessarily coming from a Dockerfile, or from the image's metadata? (didn't look at the code)
Reason I'm asking: Container images may be built without a Dockerfile

Reply:
They are coming from both depending on the flow that is being used.

* For auto detected user instructions flow, they come from the latest 5 hours worth of `history.created_by` https://github.com/opencontainers/image-spec/blob/master/config.md)
* For the dockerfile flow, they come from the run statements of the dockerfile the user provided